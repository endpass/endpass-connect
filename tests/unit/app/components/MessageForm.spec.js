import { shallowMount, mount } from '@vue/test-utils';
import MessageForm from '@/components/MessageForm.vue';

describe('MessageForm', () => {
  describe('render', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallowMount(MessageForm, {
        propsData: {
          message: 'foo',
        },
      });
    });

    it('should correctly render MessageForm component', () => {
      expect(wrapper.name()).toBe('MessageForm');
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('behavior', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = mount(MessageForm, {
        propsData: {
          message: 'foo',
        },
      });
    });

    describe('cancel feature', () => {
      it('should emit close on click cancel button by default', () => {
        wrapper.find('[data-test=cancel-button]').trigger('click');

        expect(wrapper.emitted().cancel).toBeTruthy();
      });

      it('should not emit close on click cancel button if closable is false', () => {
        wrapper.setProps({
          closable: false,
        });

        wrapper.find('[data-test=cancel-button]').trigger('click');

        expect(wrapper.emitted().cancel).toBeFalsy();
      });
    });
  });
});
