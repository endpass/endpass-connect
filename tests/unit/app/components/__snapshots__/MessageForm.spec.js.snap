// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageForm render should correctly render MessageForm component 1`] = `
<div>
  <form-field-stub>
    <message-stub data-test="form-message">foo</message-stub>
  </form-field-stub>
  <form-controls-stub>
    <v-button-stub data-test="cancel-button">Close</v-button-stub>
  </form-controls-stub>
</div>
`;
